### ISSUE TITLE
Possible bug with left shifting of signed value in source code

### ISSUE THREAD

**Issue by j123123** [CVC4/CVC4#2844]
This function https://github.com/CVC4/CVC4/blob/549fe66e9cd274784edac47203b832ff7797834f/src/expr/pickler.cpp#L75-L79
looks ambiguos. If you sure that each variable ``(char a, char b, char c, char d)`` is positive, it's fine. But shifting a negative signed value is undefined.

For example, if you do something like
```c++
std::cout <<  (unsigned)((-1) << 1) << std::endl;
```
you get 4294967294 in output and warning from compiler:
```
prog.cc:6:33: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
  std::cout <<  (unsigned)((-1) << 1) << std::endl;
                           ~~~~ ^
```

On some platform char can be unsigned, but you should not rely on it.

**Comment by 4tXJ7f** [comment]
Fortunately, this code is not very critical but we should still address this. Thank you for bringing this up!

**Comment by ajreynol** [comment]
Closed by #3236 (the code has been deleted).
